src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -214,6 +214,40 @@ bool TryConvertEnumerable(object value, Type type, Destructuring destructuring,
                 {
                     result = SequenceValue.Empty;
                 }
+                else if (list is Array a && a.Rank > 1)
+                {
+                    var rows = new LogEventPropertyValue[a.GetLength(0)];
+                    if (a.Rank == 2)
+                    {
+                        for (int i = 0; i < rows.Length; ++i)
+                        {
+                            var columns = new LogEventPropertyValue[a.GetLength(1)];
+                            for (int j = 0; j < columns.Length; ++j)
+                                columns[j] = _depthLimiter.CreatePropertyValue(a.GetValue(i, j), destructuring);
+                            rows[i] = new SequenceValue(columns);
+                        }
+                    }
+                    else if (a.Rank == 3)
+                    {
+                        for (int i = 0; i < rows.Length; ++i)
+                        {
+                            var columns = new LogEventPropertyValue[a.GetLength(1)];
+                            for (int j = 0; j < columns.Length; ++j)
+                            {
+                                var heights = new LogEventPropertyValue[a.GetLength(2)];
+                                for (int k = 0; k < heights.Length; ++k)
+                                    heights[k] = _depthLimiter.CreatePropertyValue(a.GetValue(i, j, k), destructuring);
+                                columns[j] = new SequenceValue(heights);
+                            }
+                            rows[i] = new SequenceValue(columns);
+                        }
+                    }
+                    else
+                    {
+                        throw new NotSupportedException("Serilog does not support multi-dimensional arrays with Rank > 3.");
```

Comment:
Since this is not a usage error but just a limitation, perhaps we should just use

```
result null;
return false;
```

here?

Reply:
Think this is a good candidate to get into 4.0; shall I make these final tweaks and merge this, @sungam3r?

Reply:
Thank you for the work on supporting multi-dimensional arrays. I hope this feature can be merged soon.

I believe there’s no reason not to support higher-dimensional arrays. Based on this branch, I have further improved the implementation to allow support for arrays with more dimensions: https://github.com/serilog/serilog/pull/2132

cc @sungam3r @nblumhardt